src/routes/notifications/v1/notifications.controller.ts

```diff
@@ -140,12 +175,38 @@ export class NotificationsController {
     @Param('chainId') chainId: string,
     @Param('uuid', new ValidationPipe(UuidSchema)) uuid: UUID,
   ): Promise<void> {
+    if (this.isPushNotificationV2Enabled) {
+      return await this.unregisterDeviceV2Compatible(chainId, uuid);
+    }
+
     await this.notificationsService.unregisterDevice({ chainId, uuid });
+  }
 
-    if (this.isPushNotificationV2Enabled) {
-      // Compatibility with V2
-      // @TODO Remove NotificationModuleV2 after all clients have migrated and compatibility is no longer needed.
+  private async unregisterDeviceV2Compatible(
+    chainId: string,
+    uuid: UUID,
+  ): Promise<void> {
+    try {
       await this.notificationServiceV2.deleteDevice(uuid);
+    } catch (error: unknown) {
+      if (error instanceof NotFoundException) {
+        // Do not throw a NotFound error when attempting to remove the token from the CGW,
+        // This ensures the TX service remove method is called
+      } else {
+        throw error;
+      }
+    }
+
+    try {
+      await this.notificationsService.unregisterDevice({ chainId, uuid });
+    } catch (error: unknown) {
+      // The token might already have been removed from the TX service.
+      // If this happens, the TX service will throw a 404 error, but it is safe to ignore it.
+      if (error instanceof Error) {
+        if ('code' in error && error.code !== 404) {
+          throw error;
+        }
+      }
```

Comment:
If the error is not `instanceof Error`, it is being swallowed here.

Reply:
Change in the new version
https://github.com/safe-global/safe-client-gateway/pull/2173/commits/6a9aa2f0f392ae47d6632dcfe27a6851fdcbe505